documentation/dataplane/dash-flow-api.md

```diff
@@ -0,0 +1,649 @@
+| SAI_FLOW_ENTRY_ATTR_DASH_DIRECTION         | `sai_dash_direction_t`   | Direction of the DASH flow                                   |
+| SAI_FLOW_ENTRY_ATTR_DASH_FLOW_ACTION       | `sai_dash_flow_action_t` | Action to be applied on the flow                             |
+| SAI_FLOW_ENTRY_ATTR_METER_CLASS            | `sai_uint32_t`           | Meter class for flow entry, used for traffic metering and policing |
+| SAI_FLOW_ENTRY_ATTR_IS_UNIDIRECTIONAL_FLOW | `bool`                   | Indicates if the flow is  unidirectional or bidirectional    |
+
+#### Reverse flow key
+
+When configuring a flow_entry, it can be specified whether it is unidirectional or bidirectional. If it is bidirectional, it can be designated as a reverse flow key, allowing for the rapid identification of the corresponding reverse flow. Of course, if a flow entry is initially established as unidirectional, its reverse flow can also be set up later, utilizing these attributes to link them together.
+
+| Attribute name                            | Type               | Description                                 |
+| ----------------------------------------- | ------------------ | ------------------------------------------- |
+| SAI_FLOW_ENTRY_ATTR_REVERSE_FLOW_ENI_MAC  | `sai_mac_t`        | Eni mac addr for the recerse flow           |
+| SAI_FLOW_ENTRY_ATTR_REVERSE_FLOW_IP_PROTO | `sai_uint8_t`      | IP protocol number for the reverse flow     |
+| SAI_FLOW_ENTRY_ATTR_REVERSE_FLOW_SRC_IP   | `sai_ip_address_t` | Source IP address for the reverse flow      |
+| SAI_FLOW_ENTRY_ATTR_REVERSE_FLOW_DST_IP   | `sai_ip_address_t` | Destination IP address for the reverse flow |
+| SAI_FLOW_ENTRY_ATTR_REVERSE_FLOW_SRC_PORT | `sai_uint16_t`     | L4 source port for the reverse flow         |
+| SAI_FLOW_ENTRY_ATTR_REVERSE_FLOW_DST_PORT | `sai_uint16_t`     | L4 destination port for the reverse flow    |
+
+#### Flow encap related attributes
+
+These are the related attributes of flow encapsulation.
+
+| Attribute name                                   | Type                       | Description                                                  |
+| ------------------------------------------------ | -------------------------- | ------------------------------------------------------------ |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY0_VNI                | `sai_uint32_t`             | Destination VNI in the underlay network                      |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY0_SIP                | `sai_uint32_t`             | Source IP address in the underlay network                    |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY0_DIP                | `sai_uint32_t`             | Destination IP address in the underlay network               |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY0_SMAC               | `sai_mac_t`                | Source MAC address in the underlay network                   |
```

Comment:
My general understanding is that the underlay_smac and underlay_dmac are not stored in the flow entry and are derived fresh for each packet from underlay routing lookup. Also, underlay_smac/dmac are not synced across HA partners.

Reply:
yes, you are right. we need to remove these 2 fields.

Reply:
Thanks, I have removed SAI_FLOW_ENTRY_ATTR_UNDERLAY0_SMAC and SAI_FLOW_ENTRY_ATTR_UNDERLAY0_DMAC